examples/ufo_toy.rs

```diff
@@ -0,0 +1,164 @@
+// Copyright 2019 the Runebender authors.
+
+//! A quick demonstration of loading and displaying a UFO glyph.
+
+use kurbo::{Affine, Circle, Line, BezPath, Vec2};
+use piet::{RenderContext, FillRule};
+use norad::{glyph::{PointType, Contour, ContourPoint}, Glyph};
+
+use druid_shell::platform::WindowBuilder;
+use druid_shell::win_main;
+
+use druid::widget::Widget;
+use druid::{
+    BoxConstraints, Geometry, Id, LayoutCtx, LayoutResult, PaintCtx,
+    UiMain, UiState,
+};
+
+struct GlyphEditor {
+    glyph: Glyph,
+    path: BezPath,
+    height: f32,
+}
+
+impl GlyphEditor {
+    fn new(glyph: Glyph) -> Self {
+        // assume glyph height is 1000 or 2000 units
+        let height = if glyph.outline.as_ref()
+            .map(|o| o.contours.iter()
+                 .flat_map(|c| c.points.iter().map(|p| p.y))
+                 .any(|h| h > 1000.))
+            .unwrap_or(false) { 4000. } else { 1000. };
+
+        let path = glyph.outline.as_ref().map(|o| make_path(&o.contours)).unwrap_or_default();
+        GlyphEditor { glyph, height, path }
+    }
+}
+
+impl Widget for GlyphEditor {
+    fn paint(&mut self, ctx: &mut PaintCtx, geom: &Geometry) {
+
+        let baseline = (geom.size.1  * 0.66) as f64;
```

Comment:
I want to get rid of the f32/f64 casting. Kinda leaning towards just making everything f64.

Reply:
yea was bumping into this.  I also think it might make sense to do things like `impl Into<Vec2> for (f32, f32)`.